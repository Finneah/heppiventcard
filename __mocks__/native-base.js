jest.mock('native-base', () => {
  return {
    StyleProvider: jest.fn().mockImplementation((x) => x),
  };
});
