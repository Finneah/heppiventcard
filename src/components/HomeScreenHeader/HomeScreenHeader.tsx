import * as React from 'react';
import {Linking, StyleSheet} from 'react-native';
import {strings} from '../../locale/i18n';

import {
  HStack,
  Pressable,
  Image,
  StatusBar,
  Heading,
  Icon,
  useColorModeValue,
  useColorMode,
} from 'native-base';

import Ionicons from 'react-native-vector-icons/Ionicons';

/**
 * @category Component
 * @namespace HomeScreenHeader
 * @description Header for HomeScreen
 */
const HomeScreenHeader: React.FC = () => {
  const {colorMode} = useColorMode();
  let onPressLogo = async (link: string): Promise<void | false> => {
    try {
      if (link && Linking.canOpenURL(link)) {
        await new Promise<void>((resolve) => resolve(Linking.openURL(link)));
      } else {
        throw new Error('Cant open Link');
      }
    } catch (error) {
      console.log(error.message);
      return false;
    }
  };
  util.onPressLogo = onPressLogo;

  return (
    <>
      <StatusBar
        backgroundColor="white"
        barStyle={colorMode === 'light' ? 'dark-content' : 'light-content'}
      />

      <HStack
        safeAreaTop
        bg={useColorModeValue('white', 'primary.800')}
        px={1}
        marginBottom={10}
        justifyContent="space-between"
        alignItems="flex-end">
        <Icon
          as={<Ionicons name="menu" />}
          color={useColorModeValue('primary.600', 'secondary.400')}
        />
        <Heading size="md">{strings('APPTITLE')}</Heading>
        <Pressable
          testID="pressableLogo"
          onPress={() => onPressLogo('http://heppivents.de/')}>
          <Image
            alt="heppiventlogo, click to open the website"
            source={require('../../image/logo.jpg')}
            style={styles.logoImage}
          />
        </Pressable>
      </HStack>
    </>
  );
};

const styles = StyleSheet.create({
  logoImage: {
    width: 120,
    height: 50,
    resizeMode: 'stretch',
    justifyContent: 'center',
  },
});
export default HomeScreenHeader;
export const util: any = {onPressLogo: null};
