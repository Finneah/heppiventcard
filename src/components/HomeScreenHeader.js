import {Header, Left, Right, Title} from 'native-base';
import React from 'react';
import {Image, Pressable, StyleSheet} from 'react-native';
import {strings} from '../i18n';
import logo from '../image/logo.gif';
import GlobalColors from '../styles/GlobalColors';
const HomeScreenHeader = ({}) => {
  return (
    <Header>
      <Left>
        <Pressable>
          <Image source={logo} style={styles.logoImage} />
        </Pressable>
      </Left>
      <Right>
        <Title style={{color: GlobalColors.dark}}>{strings('APPTITLE')}</Title>
      </Right>
    </Header>
  );
};
const styles = StyleSheet.create({
  logoImage: {
    backgroundColor: GlobalColors.lightGrey,
    width: 120,
    height: 50,
    resizeMode: 'stretch',
    justifyContent: 'center',
  },
});
export default HomeScreenHeader;
