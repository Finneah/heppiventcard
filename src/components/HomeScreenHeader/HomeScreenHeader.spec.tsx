import {NativeBaseProvider} from 'native-base';
import * as React from 'react';

import {render, fireEvent} from '@testing-library/react-native';

import HomeScreenHeader, {util} from './HomeScreenHeader';

const wrapper = ({children}: any) => (
  <NativeBaseProvider
    initialWindowMetrics={{
      frame: {x: 0, y: 0, width: 0, height: 0},
      insets: {top: 0, left: 0, right: 0, bottom: 0},
    }}>
    {children}
  </NativeBaseProvider>
);

describe('HomeScreenHeader', () => {
  test('renders correctly', () => {
    const homeScreenHeader = render(<HomeScreenHeader />, {wrapper});
    expect(homeScreenHeader.toJSON()).toMatchSnapshot();
  });

  test('Header Title ', () => {
    const {getByText} = render(<HomeScreenHeader />, {wrapper});
    const header = getByText('APPTITLE');
    expect(header).toBeTruthy();
  });

  test('Pressable fireEvent correctly ', () => {
    const {queryByTestId} = render(<HomeScreenHeader />, {wrapper});
    const foundElement = queryByTestId('pressableLogo');
    expect(foundElement).toBeTruthy();
    if (foundElement) {
      fireEvent.press(foundElement);
    }
  });

  test('onPressLogo function works correctly', async () => {
    render(<HomeScreenHeader />, {wrapper});

    expect(await util.onPressLogo('http://heppivents.de/')).toBeUndefined();

    expect(await util.onPressLogo()).toBeFalsy();
  });
});
