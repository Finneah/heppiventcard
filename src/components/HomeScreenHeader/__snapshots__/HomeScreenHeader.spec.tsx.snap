// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreenHeader renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    alignItems="flex-end"
    bg="white"
    flexDirection="row"
    justifyContent="space-between"
    marginBottom={10}
    px={2}
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 40,
          "paddingLeft": 8,
          "paddingRight": 8,
        },
      ]
    }
  >
    <Text
      color="muted.700"
      fontFamily="body"
      fontSize="xl"
      fontStyle="normal"
      fontWeight="bold"
      style={
        Array [
          Object {
            "color": "#404040",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "700",
          },
        ]
      }
    >
      APPTITLE
    </Text>
    <View
      style={
        Array [
          Object {},
        ]
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onHoverIn={[Function]}
      onHoverOut={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {},
        ]
      }
      testID="pressableLogo"
    >
      <Image
        accessibilityLabel="heppiventlogo, click to open the website"
        alt="heppiventlogo, click to open the website"
        maxWidth="100%"
        onError={[Function]}
        source={
          Object {
            "testUri": "../../../src/image/logo.jpg",
          }
        }
        style={
          Array [
            Object {
              "maxWidth": "100%",
            },
            Object {
              "height": 50,
              "justifyContent": "center",
              "resizeMode": "stretch",
              "width": 120,
            },
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
